### DISCUSSION TITLE
Compiling Windows executables on a Mac

### DISCUSSION THREAD

**Discussion by drdrecad** [unoplatform/uno#9475] [General]
Dear all,
we are working on a multiplatform project and we have our development machines running on MacOS. However, the ultimate target system is a Windows machine. Apart from doing this the traditional way, i.e. by running a VM, installing Windows and compiling, I was wondering if we could use Uno to simply build the executables directly from our MacOS environment, as that would save us a lot of time.

Please let me know what your thoughts are.

BR

**Comment by jeromelaban** [comment]
Could you tell which projects heads you want to run on Windows? Is it the `.Windows` head or another (like the `.Gtk` one)?

**Comment by drdrecad** [comment]
Hi Jerome,
I am not quite sure how to answer your question.

The idea is to simply be able to compile a Windows Forms or WPF project directly on MacOS, not necessarily run it there.

BR

**Reply by jeromelaban** [reply]
Windows Forms is not a project type that Uno handles directly, so I cannot tell. WPF _may_ be buildable on mac, but there may be tricks there. In all cases, you should be able to rely on a CI environment to provide you this kind of binaries.

I was asking about the `.Windows` project because it is a WinUI/WinAppSDK project, and this type of project does not build on mac.
